src/pages/home/report/ReportActionsView.js

```diff
@@ -46,8 +62,15 @@ class ReportActionsView extends React.Component {
         this.scrollToListBottom = this.scrollToListBottom.bind(this);
         this.recordMaxAction = this.recordMaxAction.bind(this);
         this.onVisibilityChange = this.onVisibilityChange.bind(this);
-        this.sortedReportActions = this.updateSortedReportActions();
+        this.setUpUnreadActionIndicator = this.setUpUnreadActionIndicator.bind(this);
+        this.updateSortedReportActions = this.updateSortedReportActions.bind(this);
+
+        this.sortedReportActions = [];
         this.timers = [];
+        this.unreadActionCount = 0;
```

Comment:
Why are we creating another variable to track the unread count, it's already passed as a prop on report, right?

I think this over complicates things a little since then we have a number we pass as a prop which we just copy to a different variable but don't really edit in any way. 

Can we just use `props.report.unreadActionCount` instead everywhere?

Reply:
We can't because the `props.report.unreadActionCount` quickly changes to 0 before the indicator fades out, which causes the indicator to disappear due to the conditional render.

Reply:
ah I see I see thanks for pointing that out. 

Looking into it a bit more looks like it happens because we call `updateLastReadActionID` here: https://github.com/Expensify/Expensify.cash/blob/89cb71597d84b62d4bc68b61bcf5fb798578f985/src/pages/home/report/ReportActionsView.js#L253

Which then in here: https://github.com/Expensify/Expensify.cash/blob/89cb71597d84b62d4bc68b61bcf5fb798578f985/src/libs/actions/Report.js#L653 calls `setLocalLastRead` and leads us to https://github.com/Expensify/Expensify.cash/blob/89cb71597d84b62d4bc68b61bcf5fb798578f985/src/libs/actions/Report.js#L212-L217 and eventually just sets `unreadActionCount` to 0.

I'm def not well versed enough here and nothing jumps out at me to make this better so 👍

Reply:
Just passing through but might be another opportunity to leave a comment explaining why we are creating a new reference. This wasn't very clear to me.

Reply:
You might be able to simplify this by making a function like this:

```javascript
shouldShowUnreadActionIndicator(index) {
    return this.isAnimating
        || (this.props.report.unreadActionCount > 0 && index === (this.props.report.unreadActionCount - 1));
}
```

Then in `renderItem` do:

```javascript
{this.shouldShowUnreadActionIndicator(index) && (
    <UnreadActionIndicator animatedOpacity={this.unreadIndicatorOpacity} />
)}
```

And in `componentDidUpdate` run:

```javascript
if (prevProps.report.unreadActionCount === 0 && this.props.report.unreadActionCount > 0) {
    this.unreadIndicatorOpacity = new Animated.Value(1);
    this.timers.push(setTimeout(() => {
        this.isAnimating = true;
        Animated.timing(this.unreadIndicatorOpacity, {
             toValue: 0,
             useNativeDriver: false,
        }).start(() => this.isAnimating = false);
    }, 3000));
}
```

That way we wouldn't need `this.unreadActionCount` and `this.props.report.unreadActionCount`. I feel like having both is maybe a bit confusing. If you didn't want to have the `isAnimating` property, you could even just check if `this.unreadIndicatorOpacity._value > 0`

Reply:
@roryabraham I'm really having a hard time figuring out a way to make it work without storing the `unreadActionCount` somewhere.
In your suggestion, the function 
```
shouldShowUnreadActionIndicator(index) {
    return this.isAnimating
        || (this.props.report.unreadActionCount > 0 && index === (this.props.report.unreadActionCount - 1));
}
```
ends up returning `true` for every action when the animation starts, which causes the indicator to show up everywhere.